torch/_inductor/scheduler.py

```diff
@@ -1332,6 +1333,68 @@ def prune_redundant_deps(
             node.prune_redundant_deps(name_to_fused_node)
 
 
+class GroupedSchedulerNode(BaseSchedulerNode):
+    """
+    This is a "fake" scheduler node that represents a group of scheduler nodes
+    that are meant to be *grouped* together (it does not allow another node to be scheduled
+    in between its constituent nodes, nor does it allow another node to fuse into any of its constituent nodes).
+    The way it does this is by maintaining its unmet dependencies as the union of its constituent nodes.
+    Fusion will still happen among the nodes within each GroupedSchedulerNode.
+    At codegen time, this scheduler node will be unpacked and codegen is called on each constituent node.
+    """
+
+    @classmethod
+    def create(cls, snodes: List[BaseSchedulerNode]) -> GroupedSchedulerNode:
+        scheduler = snodes[0].scheduler
+        assert all(node.scheduler is scheduler for node in snodes)
+        return cls(scheduler, snodes)  # type: ignore[arg-type]
+
+    def __init__(self, scheduler: Scheduler, snodes: List[BaseSchedulerNode]) -> None:
+        # NB: No need to call super().__init__() because we don't need to re-use any of its logic.
+
+        assert not any(isinstance(x, FusedSchedulerNode) for x in snodes)
+        self.snodes = snodes
+        self.scheduler = scheduler
+        for snode in snodes:
+            scheduler.name_to_fused_node[snode.get_name()] = self
+        self.node = None
+        self.ancestors = set.union(
+            *[x.ancestors for x in snodes if x.ancestors is not None]
+        )
+
+        self.set_read_writes(
+            dependencies.ReadWrites.merge_list([x.read_writes for x in snodes])
+        )
+
+        self.unmet_dependencies = {
+            dep
+            for dep in set.union(*[x.unmet_dependencies for x in snodes])
+            if dep.name not in self.get_buffer_names()
+        } - self.read_writes.writes
+
+        self.min_order = sys.maxsize
+        self.max_order = -sys.maxsize
```

Comment:
This is all copy-pasta'd from FusedSchedulerNode. Should we subclass, or can we factor out the logic?

Reply:
factored out into a common function: `init_group_node`